packages/client/src/components/UploadForm.tsx

```diff
@@ -157,22 +163,24 @@ export const UploadForm = () => {
 	const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
 		event.preventDefault();
 
-		const maybeFileInput = document.querySelector(
-			'input[id=files]',
-		) as HTMLInputElement;
-		if (
-			!maybeFileInput ||
-			!maybeFileInput.files ||
-			maybeFileInput.files.length === 0
-		) {
-			setErrorMessage(
-				'Invalid file input - did you select a file to transcribe?',
-			);
-			setStatus(RequestStatus.Failed);
+		// The react Select components with a required property don't show any
+		// feedback when the form is submitted without an option having been
+		// chosen. We need to validate that input manually.
+		if (mediaFileLanguageCode === undefined) {
+			setLanguageCodeValid(false);
 			return;
 		}
+
+		// the required property on the file input should prevent the form from
+		// being submitted without any files selected. Need to confirm this in
+		// order to narrow the type of files
+		if (files === null || files.length === 0) {
+			console.error('form submitted without any files');
```

Comment:
since we don't have anything shipping client side logs anywhere, I'm not sure it's worth logging here